src/GroupManager.php

```diff
@@ -121,13 +137,59 @@ public function removeGroup($entity_type_id, $bundle_id) {
       $editable->set('groups', $groups);
       $saved = $editable->save();
 
+      // Remove all roles associated with this group type.
+      $this->removeRoles($entity_type_id, $bundle_id);
+
       $this->refreshGroupMap();
 
       return $saved;
     }
   }
 
   /**
+   * Creates default roles for the given group type.
+   *
+   * @param string $entity_type_id
+   *   The entity type ID of the group for which to create default roles.
+   * @param string $bundle_id
+   *   The bundle ID of the group for which to create default roles.
+   */
+  protected function createDefaultRoles($entity_type_id, $bundle_id) {
```

Comment:
this should actually be `createRoles` - it should create the default (undeletable) ones, and the optional ones.